python/pysrc/kaskada/_timestream.py

```diff
@@ -65,6 +67,63 @@ def _literal(value: Literal, session: _ffi.Session) -> Timestream:
             raise TypeError("Cannot create a literal Timestream from a datetime")
         return Timestream(_ffi.Expr.literal(session, value))
 
+    @staticmethod
+    def _call_udf(
+        udf: _ffi.Udf,
+        *args: Union[Timestream, Literal],
+        session: Optional[_ffi.Session] = None,
+    ) -> Timestream:
+        """
+        Construct a new Timestream by calling the given function.
+
+        Parameters
+        ----------
+        func : str
+            Name of the function to apply.
+        *args : Timestream | int | str | float | bool | None
+            List of arguments to the expression.
+        session : FFI Session
+            FFI Session to create the expression in.
+            If unspecified, will infer from the arguments.
+            Will fail if all arguments are literals and the session is not provided.
+
+        Returns
+        -------
+        Timestream
+            Timestream representing the result of the function applied to the arguments.
+
+        Raises
+        ------
+        # noqa: DAR401 _augment_error
+        TypeError
+            If the argument types are invalid for the given function.
+        ValueError
+            If the argument values are invalid for the given function.
+        """
+        if session is None:
+            session = next(
+                arg._ffi_expr.session() for arg in args if isinstance(arg, Timestream)
+            )
+
+        def make_arg(arg: Union[Timestream, Literal]) -> _ffi.Expr:
```

Comment:
I wonder if this can either use more of the code from `Expr.call`? Isn't this the same as `call` but the name is now `str | Udf`?